NEWS.md

```diff
@@ -1,60 +1,55 @@
-3.23.1
+3.24.1
 ======
 
+## Highlights
 
-## Features
-
- * The `redis()` destination now handles any number of command parameters.
-   (#2816)
-
- * The `format()` option of `date-parser()` supports lists. From now on, a
-   single `date-parser()` instance is able to process different date formats,
-   making it easy to catch on when some programs change the way they log
-   information. (#2779)
+ * Add a new template function called `format-flat-json()`, which generates
+   flattened json output. This is useful for destinations, where the json
+   parsing does not handle nested json format. (#2890)
+ * Add ISO 8601 compliant week numbering. Use it with the `${ISOWEEK}` macro
+   and the timestamp prefixes `P_, S_, R_, C_` (#2878)
+ * Add `add-contextual-data()` glob selector. It matches the message with shell
+   style globbing. Enable it by setting `selector(glob("$my_template")` in the
+   `add-contextual-data()` block. (#2936)
+ * Send Server Name Identification (SNI) information with `transport(tls)`.
+   Enable it by setting the `sli(yes)` option in the `tls` block in your
+   `destination`. (#2930)
 
- * Add relocation support for `disk-buffer()`. The `relocate` subcommand of
-   `dqtool` can be used to move a single or multiple queue files. (#2855)
-
- * `file(), pipe()`: The `time-reap()` option now can be set or disabled for
-   each destination separately.
-   Use `time-reap(0)` to disable closing idle destination files. (#2798)
+## Features
 
- * `syslog-ng-ctl`: Add `list-files` subcommand to print files present in the
-   current configuration. (#2797)
+ * Timezone related `rewrite` rules (#2818)
+ * `templates`: change the `$LOGHOST` macro to honour `use-fqdn()` (#2894)
+ * `syslog-ng-sysconfdir` define (#2932)
+ * Dqtool assign to persist (#2872)
 
 ## Bugfixes
 
- * Fix minor memory leaks (#2868)
- * Add global context to standalone parsers (#2876)
- * Fix heap usage after free in cfg_run_parser_with_main_context (#2884)
- * Fix g_thread_init call order issue with older glibs (#2853)
- * SNMP destination: fixing statistics format (#2854)
- * eventlog: flush escaped_buffer when full (#2837)
- * dbparser: remove unnecessary lock (#2838)
- * dbparser: fix crash when context times out in the middle of another rule (#2832)
- * radix: fix grouping in PCRE (#2808)
- * add-contextual-data: make filters config plugin aware (#2886)
- * Undefined warning regression (#2829)
+ * Fix backtick substitution (#2906, #2909)
+ * Fix SCL block parameter substitution  (#2901)
+ * `logthrsource/fetcher`: crash after failed reload (#2907)
+ * Fix crash at shutdown on 32bit systems (#2893, #2895)
+ * Invalidate `LEGACY_MSGHDR` when `PID` or `PROGRAM` is `unset` (#2896)
```

Comment:
Invalidate the value of the LEGACY_MSGHDR macro in case either the PID or the PROGRAM macros are unset() using a rewrite rule. Previously LEGACY_MSGHDR would retain the old values.

Reply:
Thanks fixed! :)